host/app/lib/schema-analysis-plugin.ts

```diff
@@ -0,0 +1,260 @@
+import type * as Babel from '@babel/core';
+import type { types as t } from '@babel/core';
+import type { NodePath, Scope } from '@babel/traverse';
+
+interface State {
+  opts: Options;
+  insideCard: boolean;
+}
+
+export interface ExternalReference {
+  type: 'external';
+  module: string;
+  name: string;
+}
+
+export type CardReference =
+  | ExternalReference
+  | {
+      type: 'internal';
+      classIndex: number;
+    };
+
+export interface PossibleCardClass {
+  super: CardReference;
+  localName: string | undefined;
+  exportedAs: string | undefined;
+  path: NodePath<t.ClassDeclaration>;
+  possibleFields: Map<string, PossibleField>;
+}
+
+export interface PossibleField {
+  card: CardReference;
+  type: ExternalReference;
+  decorator: ExternalReference;
+}
+
+export interface Options {
+  possibleCards: PossibleCardClass[];
+}
+
+export function schemaAnalysisPlugin(_babel: typeof Babel) {
+  // let t = babel.types;
+  return {
+    visitor: {
+      ClassDeclaration: {
+        enter(path: NodePath<t.ClassDeclaration>, state: State) {
+          if (!path.node.superClass) {
+            return;
+          }
+
+          let sc = path.get('superClass');
+          if (sc.isReferencedIdentifier()) {
+            let cardRef = makeCardReference(path.scope, sc.node.name, state);
+            if (cardRef) {
+              state.insideCard = true;
+              let exportedAs: string | undefined;
+              let { parentPath } = path;
+              let localName = path.node.id ? path.node.id.name : undefined;
+              if (parentPath.isExportNamedDeclaration()) {
+                // the class declaration is part of a named export
+                exportedAs = localName;
+              } else if (parentPath.isExportDefaultDeclaration()) {
+                // the class declaration is part of a default export
+                exportedAs = 'default';
+              } else {
+                // the class's identifier is referenced in a node whose parent is an ExportSpecifier
+                let binding = localName
+                  ? path.scope.getBinding(localName)
+                  : undefined;
+                if (binding) {
+                  let maybeExportSpecifierLocal = binding.referencePaths.find(
+                    (b) => b.parentPath?.isExportSpecifier()
+                  ) as NodePath<t.Identifier> | undefined;
+                  if (maybeExportSpecifierLocal) {
+                    exportedAs = getName(
+                      (
+                        maybeExportSpecifierLocal.parentPath as NodePath<t.ExportSpecifier>
+                      ).node.exported
+                    );
+                  }
+                }
+              }
```

Comment:
The schema inspector forced the issue in terms of exposing the `exportedAs` information for the cards, as the schema inspector needs to be able to show the card instance creation form for the cards that are able to have `CardJSON` created for them. 

The approach I took was to looks for 3 specific scenarios:
1. the class declaration is part of the export itself, in which case I just check to see if the parent of the class declaration is an ExportNamedDeclaration.
2. the class declaration is a default export, similar to the named export declaration except that the parent will be an ExportDefaultDeclaration
3. the class's identifier is part of an export specifier. In this case I get the reference paths for the class's identifier and check to see if any of them are within an ExportSpecifier.